.github/actions/createOrUpdateStagingDeploy/createOrUpdateStagingDeploy.js

```diff
@@ -4,66 +4,147 @@ const moment = require('moment');
+            );
+
+            return GithubUtils.generateStagingDeployCashBody(
+                tag,
+                _.pluck(PRList, 'url'),
+                _.pluck(_.where(PRList, {isVerified: true}), 'url'),
+                _.pluck(deployBlockers, 'url'),
+                _.pluck(_.where(deployBlockers, {isResolved: true}), 'url'),
+            );
+        })
+        .then((body) => {
+            const defaultPayload = {
                 owner: GithubUtils.GITHUB_OWNER,
                 repo: GithubUtils.EXPENSIFY_CASH_REPO,
-                labels: GithubUtils.STAGING_DEPLOY_CASH_LABEL,
-                state: 'closed',
+                body,
+            };
+
+            if (shouldCreateNewStagingDeployCash) {
+                return GithubUtils.octokit.issues.create({
+                    ...defaultPayload,
+                    title: `Deploy Checklist: Expensify.cash ${moment().format('YYYY-MM-DD')}`,
+                    labels: [GithubUtils.STAGING_DEPLOY_CASH_LABEL],
+                    assignees: [GithubUtils.APPLAUSE_BOT],
+                });
+            }
+
+            return GithubUtils.octokit.issues.update({
+                ...defaultPayload,
+                issue_number: currentStagingDeployCashIssueNumber,
             });
-        }
-
-        // Unexpected error!
-        console.error('Unexpected error occurred finding the StagingDeployCash!'
-                + ' There may have been more than one open StagingDeployCash found,'
-                + ' or there was some other problem with the Github API request.', err);
-        core.setFailed(err);
-    })
-    .then((githubResponse) => {
-        if (!githubResponse || !githubResponse.data || _.isEmpty(githubResponse.data)) {
-            console.error('Failed fetching data from Github!', githubResponse);
-            throw new Error('Failed fetching data from Github');
-        }
-
-        // Parse the tag from the most recent StagingDeployCash
-        const lastTag = GithubUtils.getStagingDeployCashData(githubResponse.data[0]).tag;
-        console.log('Found tag of previous StagingDeployCash:', lastTag);
-
-        // Find the list of PRs merged between the last StagingDeployCash and the new version
-        return GitUtils.getPullRequestsMergedBetween(lastTag, newVersion);
-    })
-    .then(PRNumbers => GithubUtils.createNewStagingDeployCash(
-        `Deploy Checklist: Expensify.cash ${moment().format('YYYY-MM-DD')}`,
-        newVersion,
-        _.map(PRNumbers, GithubUtils.getPullRequestURLFromNumber),
-    ))
-    .then(({data}) => console.log('Successfully created new StagingDeployCash! 🎉', data.html_url))
-    .catch((err) => {
-        console.error('An error occurred!', err);
-        core.setFailed(err);
-    });
+        })
+        .then(({data}) => {
+            console.log(
+                `Successfully ${shouldCreateNewStagingDeployCash ? 'created new' : 'updated'} StagingDeployCash! 🎉`,
+                data.html_url,
```

Comment:
Why the mix between template strings and `console.log` arguments ?

Reply:
Meh, no method to the madness here. This was prettymuch an arbitrary decision, but I can clean it up!

Reply:
Updated 👍